packages/front-end/components/Features/SDKEndpoints.tsx

```diff
@@ -93,71 +94,60 @@ const SDKEndpoints: FC<{
                       )}
                     </Tooltip>
+                    <span style={{ fontSize: "87.5%", fontStyle: "italic" }}>
+                      {key.description}
+                    </span>
                   </td>
                   <td>
-                    <code>{endpoint}</code>{" "}
-                    <FaCopy
-                      className="cursor-pointer"
-                      title="Copy to Clipboard"
+                    <Tooltip
+                      role="button"
+                      tipMinWidth="45px"
+                      tipPosition="top"
+                      body={
+                        currentCopiedString !== endpoint ? "Copy" : "Copied!"
+                      }
                       onClick={(e) => {
                         e.preventDefault();
                         navigator.clipboard
                           .writeText(endpoint)
                           .then(() => {
-                            console.log("Copied!");
+                            setCurrentCopiedString(endpoint);
                           })
                           .catch((e) => {
                             console.error(e);
                           });
                       }}
-                    />
+                    >
+                      {endpoint}
+                    </Tooltip>
                   </td>
                   {hasEncryptedEndpoints && (
                     <td>
-                      {key.encryptSDK ? (
-                        canManageKeys ? (
-                          <ClickToReveal
-                            valueWhenHidden="Reveal key"
-                            getValue={async () => {
-                              const res = await apiCall<{
-                                key: ApiKeyInterface;
-                              }>(`/keys/reveal`, {
-                                method: "POST",
-                                body: JSON.stringify({
-                                  id: key.id,
-                                }),
-                              });
-                              if (!res.key?.encryptionKey) {
-                                throw new Error(
-                                  "Could not load encryption key"
-                                );
-                              }
-                              return res.key.encryptionKey;
-                            }}
-                          >
-                            {(value) => <CopyToClipboard text={value} />}
-                          </ClickToReveal>
-                        ) : (
-                          "yes"
-                        )
+                      {canManageKeys && key.encryptSDK ? (
+                        <ClickToReveal
+                          rowReverse
+                          keyId={key.id}
+                          currentCopiedString={currentCopiedString}
+                          setCurrentCopiedString={setCurrentCopiedString}
```

Comment:
Does this work?  It looks like the `ClickToReveal` component specifically uses the `key.key` property as the value instead of the encryption key.  I like the old style generic props better where the ClickToReveal component doesn't make any assumptions about how the secret data is fetched.

Reply:
Yea - the more I look at it now, the more I agree. I've just refactored the component to accept a getValue prop like it had before.